Building Clouds

MassterKek opened the pull request:
Still having an issue getting the circles to appear, can't figure out why they won't show.

MassterKek committed:
Nothing appears in the DOM

Still having an issue getting the circles to appear, can't figure out why they won't show.

MassterKek committed:
Got the circles to appear

cronin4392 commented on js/site.js:
```diff
@@ -49,6 +48,56 @@ function draw() {
+  const vector_max = rad + pre_pos.rad;
+  const vector_size = remapNumber(Math.random(), 0, 1, vector_min, vector_max);
+  const vector_angle = remapNumber(Math.random(), 0, 1, 225, 315);
+  const xpos = Math.cos(vector_angle)*vector_size;
```
I think you have x and y flipped. I think x should be sin and y cos?

MassterKek committed:
[WIP] Set up Cloud SVG building function, moving to animation

Created the cloudSVG function which returns an svg element containing randomly positioned circles.
Currently working on animating and coloring the clouds based on weather data.

MassterKek committed:
[WIP] Trying to make shade of gray in clouds based on precipitation

cronin4392 commented:
Looking good, approved the PR. @MassterKek you can merge whenever you're ready. I'd recommend doing "Squash and merge" from the merge button dropdown.

MassterKek committed:
Finished building clouds

Next step is to try to stagger the animation of each circle in the svg

cronin4392 commented on js/site.js:
```diff
@@ -78,31 +79,33 @@ function cloudSVG(n, precip) {
   let c = remapNumber(n, 0, 200, 25, 85);
   let col_val = 110 - c;
-  return `hsl(0%, 0%, ${col_val}%)`;
+  console.log(`hsl(0, 0%, ${col_val}%)`);
```
Out of curiosity @MassterKek, are you using these console logs for debugging? If so, I'd keep them, but if you're just adding out of habit I'd recommend removing them. Just adds noise.

MassterKek replied:
They were for debugging, was going to clear them later

cronin4392 commented on js/site.js:
```diff
@@ -67,7 +68,7 @@ function generatePositionValues(pre_pos) {
 
 // Creates an svg containing n circles
-function cloudSVG(n, precip) {
+function cloudSVG(n, p) {
```
I personally tend to try and use somewhat descriptive variable names. Helps with debugging and comprehending the code later.

MassterKek replied:
Noted, will adjust

cronin4392 commented:
When you merge it, just remember to take out the [WIP] from the title!

MassterKek committed:
Cleaned up Javascript

cronin4392 commented on css/styles.css:
```diff
@@ -1,20 +1,19 @@
 }
 
 circle {
+  position: absolute;
```
remove?

cronin4392 commented on js/site.js:
```diff
@@ -17,14 +19,16 @@ fetch(api)
       temp_c,
     } = data.current;
-
+    precip = data.current.precip_mm;
```
You can rename a variable when you pull it out of the object above, for example:

```
const { precip_mm: precip, } = data.current;
```

MassterKek committed:
[WIP] Removed id system for cloud svgs

MassterKek committed:
Fixed up cloud building functions

Ready to move onto animation.

MassterKek committed:
Minor cleanup
